src/App.jsx

```diff
@@ -14,45 +16,156 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
+const SORT_ASC = 'asc';
+const SORT_BY_LENGTH = 'length';
+
+function getPreparedGoods(goods, sortList, isReversed) {
+  const preparedGoods = [...goods];
+
+  if (sortList) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortList) {
+        case SORT_ASC:
+          return good1.localeCompare(good2);
+
+        case SORT_BY_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortList, setSortList] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+
+  const sortGoods = getPreparedGoods(goodsFromServer, sortList, isReversed);
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={`button is-info ${cN({ 'is-light': sortList !== SORT_ASC })}`}
+          onClick={() => setSortList(SORT_ASC)}
```

Comment:
it is bad practise to make arrow functions in jsx.
consider to make one funciton for each sortType which will be accept sortField and return arrow function.
```
const makeSetSortList = (field) => () => setSortList(field);
```
```suggestion
          onClick={(makeSetSortList(SORT_ASC)}
```